modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/statistic/SqlStatisticManagerImpl.java

```diff
@@ -151,6 +126,57 @@ public void start() {
                 tableSizeMap.putIfAbsent(table.id(), DEFAULT_VALUE);
             }
         }
+
+        scheduler.scheduleAtFixedRate(this::update, INITIAL_DELAY, REFRESH_PERIOD, TimeUnit.MILLISECONDS);
+    }
+
+    private void update() {
+        if (!latestUpdateFut.get().isDone()) {
+            return;
+        }
+
+        for (Map.Entry<Integer, ActualSize> ent : tableSizeMap.entrySet()) {
+            Integer tableId = ent.getKey();
+
+            if (droppedTables.contains(tableId)) {
+                continue;
+            }
+
+            TableViewInternal tableView = tableManager.cachedTable(tableId);
+
+            if (tableView == null) {
+                LOG.debug("No table found to update statistics [id={}].", ent.getKey());
+                continue;
+            }
+
+            CompletableFuture<Void> updateResult = statSupplier.estimatedSizeWithLastUpdate(tableView.internalTable())
+                    .handle((res, err) -> {
+                            if (err != null) {
+                                LOG.debug("Can't calculate size for table [id={}].", err, tableId);
```

Comment:
Error message is not informative. Let's be more specific. 

```suggestion
                                LOG.debug("Failed to update table statistics for [tableId={}].", err, tableId);
```

Reply:
done